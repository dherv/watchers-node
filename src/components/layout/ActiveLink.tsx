import { useRouter } from "next/router";
import styled from "styled-components";

const ActiveLink = ({ children, href }) => {
  const router = useRouter();

  const handleClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
    event.preventDefault();
    router.push(href);
  };

  return (
    <ListItem href={href} path={router.pathname}>
      <Anchor href={href} onClick={handleClick} height={60}>
        {children}
      </Anchor>
    </ListItem>
  );
};

const Anchor = styled.a<{ href: string; height: number }>`
  height: ${props => props.height}px;
  line-height: ${props => props.height}px;
  width: 100%;
`;

const ListItem = styled.li<{ path: string; href: string }>`
  font-weight: 100;
  font-size: 1.1rem;
  margin: 0 4rem;
  position: relative;
  &::after {
    content: "";
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    border-width: 2px;
    border-style: solid;
    border-color: ${props =>
      props.path === props.href ? "#2e8e89" : "transparent"};
  }
`;
export { Anchor, ListItem };
export default ActiveLink;
