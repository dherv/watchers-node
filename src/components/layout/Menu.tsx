import React from "react";
import ActiveLink from "./ActiveLink";
import styled from "styled-components";

const Menu = () => (
  <List>
    <ListItem>
      <ActiveLink href="/movies">movies</ActiveLink>
    </ListItem>
    <ListItem>
      <ActiveLink href="/series">series</ActiveLink>
    </ListItem>
    <ListItem>
      <ActiveLink href="/watchlist">watchlist</ActiveLink>
    </ListItem>
    <ListItem>
      <ActiveLink href="/likelist">likelist</ActiveLink>
    </ListItem>
  </List>
);

const List = styled.ul`
  display: flex;
  margin-left: 8rem;
`;

const ListItem = styled.li`
  font-weight: 100;
  font-size: 1.1rem;
  margin: 0 4rem;
`;

export { List, ListItem };
export default Menu;
