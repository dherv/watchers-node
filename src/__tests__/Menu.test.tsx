import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
//import "jest-styled-components";

import Menu, { List } from "../components/layout/Menu";
import ActiveLink from "../components/layout/ActiveLink";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: ""
    };
  }
}));

describe("Menu component", () => {
  const wrapper = shallow(<Menu />);

  describe("user interface", () => {
    test("should display one list", () => {
      expect.assertions(1);
      expect(wrapper.find(List)).toHaveLength(1);
    });

    test("should display 4 links with the correct text and href", () => {
      expect.assertions(5);
      const items = [
        { href: "/movies", children: "movies" },
        { href: "/series", children: "series" },
        { href: "/watchlist", children: "watchlist" },
        { href: "/likelist", children: "likelist" }
      ];
      const links = wrapper.find(ActiveLink);
      expect(links).toHaveLength(4);
      links.forEach((item, index) => {
        expect(item.props()).toStrictEqual(items[index]);
      });
    });
  });

  describe("snapshot", () => {
    test("should match", () => {
      expect.assertions(1);
      const wrapper = renderer.create(<Menu />).toJSON();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
