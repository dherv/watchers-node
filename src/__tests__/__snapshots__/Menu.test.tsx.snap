// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component snapshot should match 1`] = `
<ul
  className="Menu__List-sc-1iwg4vp-0 cgYyLU"
>
  <li
    className="ActiveLink__ListItem-sc-1si0yf-1 eCdVPD"
    href="/movies"
  >
    <a
      className="ActiveLink__Anchor-sc-1si0yf-0 bksUxL"
      height={60}
      href="/movies"
      onClick={[Function]}
    >
      movies
    </a>
  </li>
  <li
    className="ActiveLink__ListItem-sc-1si0yf-1 eCdVPD"
    href="/watchlist"
  >
    <a
      className="ActiveLink__Anchor-sc-1si0yf-0 bksUxL"
      height={60}
      href="/watchlist"
      onClick={[Function]}
    >
      watchlist
    </a>
  </li>
</ul>
`;
